Delicate-Butterflies/Bangazon-Site
# Sb delete product

Related Issue: #19

Description: Allows users to view their products and delete products they no longer wish to sell. 

Chai/Mocha Tests File:

PR Testing Steps:
- ```npm run teardown```
- fetch the branch
- ```npm run dbrb```
- login to a user
- look at that users products by clicking 'My Products'
- pick a product from that view and search for that product
- verify that it is a search result
- return to 'My Products' and delete that product
- search for that product again and it should no longer appear
- repeat for categories as well as a product you have created
- go into psql and verify that product is still in the database but has a 'deletedAt' column with a value 
- done

Check list before you make a PR:
